pkg/volume/csi/csi_attacher.go

```diff
@@ -166,6 +168,43 @@ func (c *csiAttacher) waitForVolumeAttachmentInternal(volumeHandle, attachID str
 	return attach.Name, nil
 }
 
+func (c *csiAttacher) waitForVolumeAttachmentWithLister(volumeHandle, attachID string, pollInterval, timeout time.Duration) error {
+	klog.V(4).Info(log("probing VolumeAttachment [id=%v]", attachID))
+
+	ticker := time.NewTicker(pollInterval)
+	timer := time.NewTimer(timeout) // TODO (vladimirvivien) investigate making this configurable
+	defer ticker.Stop()
+	defer timer.Stop()
+
+	for {
+		volumeAttachment, err := c.plugin.volumeAttachmentLister.Get(attachID)
+		if err != nil {
+			// Ignore "not found" errors in case the VolumeAttachment was just created and hasn't yet made it into the lister.
```

Comment:
maybe first wait for pollInterval before get from lister to help avoid always retrying?

Reply:
This was the original approach. [See this comment thread](https://github.com/kubernetes/kubernetes/pull/97834#discussion_r569994822) for context on why I changed it. I can always move it back if you feel this is a waste of a timer tick.

Reply:
I remember you did some experiments about this. One tick is 500 ms. You can decide which way to go based on your data?

Reply:
I tuned the ticker to be a bit more aggressive, this time 100ms. [These are the timings](https://gist.github.com/chrishenzie/2ed57c1a6bf45770c1bdffc93ee48036), compared with the [old timings when it was 500ms](https://gist.github.com/chrishenzie/237e9b06fffdbd3bbfd07c6dee121830). The speedup is somewhat negligible, around half a second to a second faster depending on test noise.

Reply:
See the comment thread below. It seems like having an aggressive frequency (like 100ms) doesn't make sense since we're waiting on the result of a storage provider's API call which may take >10s. This should be tuned to something less frequent, like maybe 1 second.

Reply:
what I mean is explicitly wait a little bit before getting to get VA from lister. I remember you mentioned that typically for the first time, this function will create VA, and then try to get VA from lister, which will fail mostly because VA has not yet in lister. Will it bring some benefit to wait a second so that lister will have the VA already when we check it here?

Reply:
also I agree 100ms is too aggressive, starting from 1 sec seems reasonable.

Reply:
Ok, I've bumped it to 1 second